runtime/src/read_only_accounts_cache.rs

```diff
@@ -23,33 +24,59 @@ type ReadOnlyCacheKey = (Pubkey, Slot);
 #[derive(Debug)]
 struct ReadOnlyAccountCacheEntry {
     account: AccountSharedData,
-    index: Index, // Index of the entry in the eviction queue.
+    /// Index of the entry in the eviction queue.
+    index: Index,
+    /// lower bits of last timestamp when eviction queue was updated
+    last_time: u32,
+}
+
+impl ReadOnlyAccountCacheEntry {
+    fn new(account: AccountSharedData, index: Index) -> Self {
+        Self {
+            account,
+            index,
+            last_time: Self::timestamp(),
+        }
+    }
+    /// lower bits of current timestamp. We don't need higher bits and u32 fits with Index u32
+    fn timestamp() -> u32 {
+        (timestamp() % (u32::MAX as u64 + 1)) as u32
```

Comment:
I was under the impression this could panic if the value could not be losslessly converted to a u32. Further research indicates I was wrong. I can simplify this to u32. I don't like it, but it looks to be how `as` [is defined.](https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions)